pkg/perf/conv.go

```diff
@@ -0,0 +1,46 @@
+package perf
+
+import "unsafe"
+
+func newStringConverter(capacity int) *stringConverter {
+	conv := stringConverter{}
+	if capacity <= 0 {
+		return &conv
+	}
+
+	conv.buf = make([]byte, 0, capacity)
+	return &conv
+}
+
+// stringConverter converts bytes to strings with less allocs.
+// The idea is to accumulate bytes in a buffer with specified capacity
+// and create strings with unsafe.String using bytes from a buffer.
+// For example, 10 "fizz" strings written to a 40-byte buffer
+// will result in 1 alloc instead of 10.
+//
+// Once a buffer is filled, strings will be allocated as usually.
+type stringConverter struct {
```

Comment:
This is just like https://github.com/parca-dev/parca-agent/blob/main/pkg/discovery/systemd/decoder.go#L205, but it doesn't allocate more buffers when the initial one fills up.